spec/forms/candidate_interface/pick_provider_form_spec.rb

```diff
@@ -4,10 +4,10 @@
   describe '#available_providers' do
     it 'returns providers with a course exposed in find in the current cycle' do
       create(:provider, name: 'School without courses')
-      create(:course, open_on_apply: true, exposed_in_find: false, provider: create(:provider, name: 'School with disabled courses'))
-      create(:course, open_on_apply: true, exposed_in_find: true, applications_open_from: Time.zone.today, provider: create(:provider, name: 'School with courses'))
-      create(:course, open_on_apply: true, exposed_in_find: true, applications_open_from: Time.zone.tomorrow, provider: create(:provider, name: 'School with courses but not open for applications'))
-      create(:course, open_on_apply: true, exposed_in_find: true, recruitment_cycle_year: RecruitmentCycle.previous_year)
+      create(:course, :open, exposed_in_find: false, provider: create(:provider, name: 'School with disabled courses'))
+      create(:course, :open, exposed_in_find: true, applications_open_from: Time.zone.today, provider: create(:provider, name: 'School with courses'))
+      create(:course, :open, exposed_in_find: true, applications_open_from: Time.zone.tomorrow, provider: create(:provider, name: 'School with courses but not open for applications'))
+      create(:course, :open, exposed_in_find: true, recruitment_cycle_year: RecruitmentCycle.previous_year)
```

Comment:
These don't actually need to be `open`. Should we change the code to filter open courses and not just exposed in find?

Reply:
The pick form filters out courses that aren't `exposed_in_find`. Should it also filter out courses that are closed by providers?

https://github.com/DFE-Digital/apply-for-teacher-training/blob/open_on_apply_candidate_interface/app/forms/candidate_interface/pick_provider_form.rb#L8-L13

Reply:
I think it should, although I assume there shouldn't be any courses that are closed and exposed in find? Although I think it should.

Reply:
The pick provider form should return providers without valid courses. The UI will handle when the candidate chooses and invalid provider and sends them to an error page. (whether that is a good idea or not.)

Reply:
We've had a discussion around the concepts of `open` and `exposed_in_find`. Also around the logic of presenting providers in this dropdown when they are not available for applications. The changes in this PR are ok to go ahead considering these discussions.